### PR TITLE
Update fwks KPIs

### PR DESCRIPTION
**Pull request by kjackiew**
# Description

- update with latest KPIs from verbose logs

# Checklist

## General

- [ ] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
- [ ] Have you formatted the code using clang-format?

### CODE REVIEW

**Comment by kealan-barbieri** [comment]
make test
set test_scope=NIGHTLY
disable test_device_cpu
enable test_device_gpu
enable arch_gpu_xe-hpc
enable arch_gpu_xe-hpg-atsm
enable arch_gpu_xe-hpg-dg2
enable arch_gpu_xe-lp
enable arch_gpu_xe-lpg
enable arch_gpu_xe-lpg+
enable arch_gpu_xe2-hpg-bmg
enable arch_gpu_xe2-lpg

**Comment by kjackiew** [comment]
make test
set test_scope=NIGHTLY
disable test_device_cpu
enable test_device_gpu
enable arch_gpu_xe-hpc
enable arch_gpu_xe-hpg-atsm
enable arch_gpu_xe-hpg-dg2
enable arch_gpu_xe-lp
enable arch_gpu_xe-lpg
enable arch_gpu_xe-lpg+
enable arch_gpu_xe2-hpg-bmg
enable arch_gpu_xe2-lpg

File: tests/benchdnn/inputs/graph/option_set_fwks_key_gpu [hunk 1]
```diff
@@ -0,0 +1,142 @@
+--reset --expected-n-partitions=1 --dt=0:f16+3:f16+4:f16+5:f16+6:f16+27:f16+50:f16 --in-shapes=0:1x16x256x1024*abcd+3:1x16x1024x256*abcd+5:1x1x1024x1024+6:1x16x1024x256*abcd --case=complex_fusion/generated/sdpa-scl-msk.json
```
Comment (dzarukin) [inline tests/benchdnn/inputs/graph/option_set_fwks_key_gpu]: `--expected-n-partitions=1` is enabled by default. Please update the infrastructure to remove it to decrease the amount of text.

File: tests/benchdnn/inputs/graph/test_graph_gpu [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Test layers of some key GPU DL Frameworks
+--reset
+--batch=option_set_fwks_key_gpu
```
Comment (dzarukin) [inline tests/benchdnn/inputs/graph/test_graph_gpu:3]: How much time does it take to test this new file?

File: tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl.json [hunk 3]
```diff
@@ -0,0 +1 @@
...f"}], "outputs": [{"id": 16, "dtype": "f32", "shape": [1, 16, 384, 384], "stride": [2359296, 147456, 384, 1], "layout_type": "strided", "property_type": "variable"}]}, {"id": 500, "name": "softmax", "kind": "SoftMax", "attrs": {"axis": {"type": "s64", "value": 3}}, "inputs": [{"id": 16, "dtype": "f32", "shape": [1, 16, 384, 384], "stride": [2359296, 147456, 384, 1], "layout_type": "strided", "property_type": "variable"}], "outputs": [{"id": 27, "dtype": "f16", "shape": [1, 16, 384, 384], "stride": [2359296, 147456, 384, 1], "layout_type": "strided", "property_type": "variable"}]}, {"id": 700, "name": "matmul_v", "kind": "MatMul", "attrs": {"transpose_a": {"type": "bool", "value": 0}, "transpose_b": {"type": "bool", "value": 0}}, "inputs": [{"id": 27, "dtype": "f16", "shape": [1, 16, 384, 384], "stride": [2359296, 147456, 384, 1], "layout_type": "strided", "property_type": "variable"}, {"id": 6, "dtype": "f16", "shape": [1, 16, 384, 64], "stride": [393216, 24576, 64, 1], "layout_type": "strided", "property_type": "variable"}], "outputs": [{"id": 50, "dtype": "f16", "shape": [1, 16, 384, 64], "stride": [393216, 24576, 64, 1], "layout_type": "strided", "property_type": "variable"}]}]}
```
Comment (dzarukin) [inline tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl.json]: Just thoughts out loud: I wonder if there's a formatter that can be used for such kind of output to make it somewhat readable and have an ability to update it if needed.

**Review by dzarukin — COMMENTED** [review]
I'm concerned that almost every update we just bring more and more test cases.

Since I'm not familiar much with the infrastructure that generates those, my question is - does the machinery do similarity analysis and filters out cases that are exact identical or really differ in minor details, such different minibatches like 5 and 6, or ic 256 versus 512.

Piling up stuff that doesn't bring more value on top of what's already there just increase the time spent on running those cases and doesn't really help getting to the goal of Nightly in 8 hours (besides other problems we need to solve to reach that).

**Comment by kealan-barbieri** [comment]
@dzarukin we are checking for exact duplicates but not for similarity, its worth investigating as a way to cut down the size of these.

File: tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl-msk.json [hunk 4]
```diff
@@ -0,0 +1 @@
...e"}], "outputs": [{"id": 18, "dtype": "f32", "shape": [1, 16, 384, 384], "stride": [2359296, 147456, 384, 1], "layout_type": "strided", "property_type": "variable"}]}, {"id": 500, "name": "softmax", "kind": "SoftMax", "attrs": {"axis": {"type": "s64", "value": 3}}, "inputs": [{"id": 18, "dtype": "f32", "shape": [1, 16, 384, 384], "stride": [2359296, 147456, 384, 1], "layout_type": "strided", "property_type": "variable"}], "outputs": [{"id": 27, "dtype": "f16", "shape": [1, 16, 384, 384], "stride": [2359296, 147456, 384, 1], "layout_type": "strided", "property_type": "variable"}]}, {"id": 700, "name": "matmul_v", "kind": "MatMul", "attrs": {"transpose_a": {"type": "bool", "value": 0}, "transpose_b": {"type": "bool", "value": 0}}, "inputs": [{"id": 27, "dtype": "f16", "shape": [1, 16, 384, 384], "stride": [2359296, 147456, 384, 1], "layout_type": "strided", "property_type": "variable"}, {"id": 6, "dtype": "f16", "shape": [1, 16, 384, 64], "stride": [393216, 24576, 64, 1], "layout_type": "strided", "property_type": "variable"}], "outputs": [{"id": 50, "dtype": "f16", "shape": [1, 16, 384, 64], "stride": [393216, 24576, 64, 1], "layout_type": "strided", "property_type": "variable"}]}]}
```
Comment (ElaineBao) [inline tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl-msk.json]: Is it intended that softmax doesn't have the `inf_as_zero` attribute? If not, I think this one should be similar to [sdpa-plain-simplified-f16.json](https://github.com/uxlfoundation/oneDNN/blob/main/tests/benchdnn/inputs/graph/complex_fusion/mha/sdpa-plain-simplified-f16.json)

File: tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl.json (hunk 3 above)
Comment (ElaineBao) [inline tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl.json]: Same question :), is it intended that softmax doesn't have the inf_as_zero attribute? If not, I think this one should be similar to [sdpa-plain-wo-mask-f16.json](https://github.com/uxlfoundation/oneDNN/blob/main/tests/benchdnn/inputs/graph/complex_fusion/mha/sdpa-plain-wo-mask-f16.json)

File: tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-dqk-scl-msk-dqv.json [hunk 5]
```diff
@@ -0,0 +1 @@
...": 3}}, "inputs": [{"id": 6, "dtype": "s8", "shape": [1, 16, 384, 64], "stride": [393216, 24576, 64, 1], "layout_type": "strided", "property_type": "variable"}, {"id": 7, "dtype": "f16", "shape": [1, 16, 384, 1], "stride": [6144, 384, 1, 1], "layout_type": "strided", "property_type": "undef"}, {"id": 8, "dtype": "s8", "shape": [1, 16, 384, 1], "stride": [6144, 384, 1, 1], "layout_type": "strided", "property_type": "undef"}], "outputs": [{"id": 45, "dtype": "f16", "shape": [1, 16, 384, 64], "stride": [393216, 24576, 64, 1], "layout_type": "strided", "property_type": "variable"}]}, {"id": 700, "name": "matmul_v", "kind": "MatMul", "attrs": {"transpose_a": {"type": "bool", "value": 0}, "transpose_b": {"type": "bool", "value": 0}}, "inputs": [{"id": 27, "dtype": "f16", "shape": [1, 16, 384, 384], "stride": [2359296, 147456, 384, 1], "layout_type": "strided", "property_type": "variable"}, {"id": 45, "dtype": "f16", "shape": [1, 16, 384, 64], "stride": [393216, 24576, 64, 1], "layout_type": "strided", "property_type": "variable"}], "outputs": [{"id": 50, "dtype": "f16", "shape": [1, 16, 384, 64], "stride": [393216, 24576, 64, 1], "layout_type": "strided", "property_type": "variable"}]}]}
```
Comment (ElaineBao) [inline tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-dqk-scl-msk-dqv.json]: Is it similar to [sdpa-compressed-kv-int4-gs32.json](https://github.com/uxlfoundation/oneDNN/blob/main/tests/benchdnn/inputs/graph/complex_fusion/mha/sdpa-compressed-kv-int4-gs32.json)?

File: tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl.json (hunk 3 above)
Comment (TaoLv) [reply tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl.json]: Test this case with benchdnn and `export ONEDNN_GRAPH_DUMP=subgraph` to let the library generate a formatted one.

File: tests/benchdnn/inputs/graph/test_graph_gpu (hunk 2 above)
Comment (kealan-barbieri) [reply tests/benchdnn/inputs/graph/test_graph_gpu:3]: test time varies based on platform, total time for `test_graph_gpu` seems to range from 2-15 minutes on platforms where it doesnt hang in CI.

File: tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl-msk.json (hunk 4 above)
Comment (kealan-barbieri) [reply tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl-msk.json]: these testcases are automatically generated from verbose logs provided by frameworks to `onednn-perf-report`, they dont appear to contain `inf_as_zero` attribute, if there is such a logfile that is being incorrectly parsed let me know.

**Comment by kealan-barbieri** [comment]
@dzarukin regarding increasing test count, the point of these updates is to improve our coverage of real fwks use cases. We can work to remove redundant or similar cases but unless models are dropped by frameworks as targets we should expect  to have marginal increases each time. Even removing similar cases isnt risk free as for example GEMM strategies can change with problem size so we will potentially lose functional testing coverage.

**Comment by dzarukin** [comment]
> @dzarukin regarding increasing test count, the point of these updates is to improve our coverage of real fwks use cases. We can work to remove redundant or similar cases but unless models are dropped by frameworks as targets we should expect to have marginal increases each time. Even removing similar cases isnt risk free as for example GEMM strategies can change with problem size so we will potentially lose functional testing coverage.

Improved coverage is a great goal, but the budget drives that. The extended coverage actually decreases the value of validation starting from a certain point (which I believe we've already reached) if one can't get results in a reasonable time frame, which creates a fascinating paradox.

While I'm cool with conv and matmul being tested more thoroughly as it's simply the library's bread and water, the rest, esp. memory-bound primitives, don't fall into that category much, given that there's a handful of different implementations that can be tested reasonably well by, let's say, 4k-5k test cases which would fit 20-45 minutes depending on a target platform instead of 2-3 hours by running 25k cases.

But even with matmul, it should be well understood what classes of kernels and strategies we are covering and what cases make sense and what not, that's why I asked about similarity, because I barely expect increasing mb size in matmul would provide any additional value to catching functional issues, or using 8k instead of 4k or 2k as N in a big shape. I'm asking to start investing into "smart coverage" since this moment as we are struggling to validate the library in a timely manner and extending coverage blindly moves us in a backward direction.

I hope everyone can read this message, recognize the effort to improve team members' lives and contribute their 2 pennies towards the goal.

**Comment by vpirogov** [comment]
@kjackiew, please reduce the scope of this PR to compute functionality only (conv, deconv, ip, matmul, graph).

File: tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl-msk.json (hunk 4 above)
Comment (ElaineBao) [reply tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl-msk.json]: Hi, @kealan-barbieri, I'm not sure if the verbose logs provided by frameworks are graph verbose logs or not? If they are graph verbose logs, the attributes of the ops are designed to be omit.. That means no matter user uses a softmax with inf_as_zero or not, the verbose log will not contain that attribute.

The only hint I can provide is that if framework is pytorch, most likely the softmax has inf_as_zero attribute. (see the [RFC](https://github.com/uxlfoundation/oneDNN/pull/2932) for more details)

**Comment by kjackiew** [comment]
make test
set test_scope=NIGHTLY
disable test_device_cpu
disable benchdnn_all
enable benchdnn_conv
enable benchdnn_deconv
enable benchdnn_graph
enable benchdnn_ip
enable benchdnn_matmul
enable test_device_gpu
enable arch_gpu_xe-hpc
enable arch_gpu_xe-hpg-atsm
enable arch_gpu_xe-hpg-dg2
enable arch_gpu_xe-lp
enable arch_gpu_xe-lpg
enable arch_gpu_xe-lpg+
enable arch_gpu_xe2-hpg-bmg
enable arch_gpu_xe2-lpg

**Comment by kjackiew** [comment]
make test
set test_scope=NIGHTLY
disable test_device_cpu
disable benchdnn_all
enable benchdnn_conv
enable benchdnn_deconv
enable benchdnn_graph
enable benchdnn_ip
enable benchdnn_matmul
enable test_device_gpu
enable arch_gpu_xe-hpc
enable arch_gpu_xe-hpg-atsm
enable arch_gpu_xe-hpg-dg2
enable arch_gpu_xe-lp
enable arch_gpu_xe-lpg
enable arch_gpu_xe-lpg+
enable arch_gpu_xe2-hpg-bmg
enable arch_gpu_xe2-lpg

File: tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl-msk.json (hunk 4 above)
Comment (atkassen) [reply tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl-msk.json]: Thanks @ElaineBao, I've updated the code that generates these files to add `inf_as_zero` as the default mode.

Comment (kealan-barbieri) [reply tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl-msk.json]: @ElaineBao you were correct this was an issue in our verbose converter that @atkassen just merged a fix for. @kjackiew if you re-generate this input file using the latest onednn-perf report this should be resolved.

File: tests/benchdnn/inputs/graph/option_set_fwks_key_gpu (hunk 1 above)
Comment (atkassen) [reply tests/benchdnn/inputs/graph/option_set_fwks_key_gpu]: I've updated the infrastructure to remove the extraneous arguments.

File: tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl.json (hunk 3 above)
Comment (atkassen) [reply tests/benchdnn/inputs/graph/complex_fusion/generated/sdpa-scl.json]: The major difference here is that these generated files use a uniform numbering for every tensor + operation:

<details><summary>ID allocations</summary>

```Python
    """
    ID allocations:
     * Operations
       - 100: K dequant
       - 200: QK GEMM
       - 300: attention scale
       - 310: causal mask row genindex
       - 320: bottom-right causal mask row add
       - 330: bottom-right causal mask row sub
       - 340: causal mask col genindex
       - 350: causal mask greater-than-or-equal
       - 400: attention mask add / causal mask select
       - 500: softmax
       - 600: V dequant
       - 700: SV GEMM
     * Tensors
       --- USER-SUPPLIED ---
       -  0: K
       -  1: K scales
       -  2: K zero points
       -  3: Q
       -  4: attention scale
       -  5: attention mask
       -  6: V
       -  7: V scales
       -  8: V zero points
       - 11: causal mask select alternative value
       - 12: bottom-right causal mask row up-shift
       - 13: bottom-right causal mask row down-shift
       --- INTERMEDIATES ---
       - 10: dequantized K
       - 15: QK
       - 16: scaled QK
       - 27: softmax result
       - 31: causal mask rows
       - 32: bottom-right causal mask up-shifted rows
       - 33: bottom-right causal mask down-shifted rows
       - 34: causal mask columns
       - 35: causal mask select mask
       - 40: causal mask select result
       - 45: dequantized V
       - 50: SV
    """
```

</details>

**Comment by kjackiew** [comment]
make test
set test_scope=NIGHTLY
disable test_device_cpu
disable benchdnn_all
enable benchdnn_graph
enable test_device_gpu
enable arch_gpu_xe-hpc
enable arch_gpu_xe-hpg-atsm
enable arch_gpu_xe-hpg-dg2
enable arch_gpu_xe-lp
enable arch_gpu_xe-lpg
enable arch_gpu_xe-lpg+
enable arch_gpu_xe2-hpg-bmg
enable arch_gpu_xe2-lpg
